<!DOCTYPE html>
<html lang="en">
<head>
    <title>Curriculum Vitae | Dr Tung Nguyen</title>
    <link rel="stylesheet"
          href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.6/dist/css/bootstrap.min.css"
          crossorigin="anonymous">
    <link rel="stylesheet"
          href="https://use.fontawesome.com/releases/v6.7.0/css/all.css"
          crossorigin="anonymous">
    <style>
        .border-bottom {
            border-bottom-color: black !important;
            border-bottom-width: 2px  !important;
            border-bottom-style: solid !important;
        }
        .title {
            text-align: left !important;
            font-weight: bolder !important;
        }
        .organisation {
            color: #0b84e0 !important;
            font-weight: bold !important;
        }
        .details {
            border-bottom-color:  black !important;
            margin-top: -15px !important;
        }
    </style>
</head>
<body>
<div class="container-fluid">
    <div class="row">
        <div class="col-md-10">
            <h3>DR TUNG NGUYEN</h3>
            <p style="color: #0b84e0">Sr. Software Engineer | Full-Stack Development | <a
                    href="https://www.ebi.ac.uk" target="_blank">EMBL-EBI</a></p>
            <p style="margin-top: -10px"><i class="fas fa-phone" style="color: #0b84e0"></i> +44 (0)7784 ABC XYZ
                &nbsp; &nbsp;
                <i class="fas fa-at" style="color: #0b84e0"></i> xxx@gmail.com &nbsp; &nbsp;
                <i class="fas fa-link" style="color: #0b84e0"></i>
                    <a href="https://linkedin.com/in/nvntung" target="_blank">LinkedIn.com</a> &nbsp; &nbsp;
                <i class="fas fa-link" style="color: #0b84e0"></i>
                <a href="https://scholar.google.com/citations?user=05M8vYEAAAAJ&hl=en&oi=sra"
                   target="_blank">Google Scholar</a> &nbsp; &nbsp;
                <i class="fas fa-link" style="color: #0b84e0"></i>
                <a href="https://github.com/ntung/" target="_blank">My GitHub</a> &nbsp; &nbsp;
                <i class="fas fa-link" style="color: #0b84e0"></i>
                <a href="https://github.com/EBI-BioModels" target="_blank">BioModels on GitHub</a> &nbsp; &nbsp;
                <i class="fas fa-link" style="color: #0b84e0"></i>
                <a href="https://bitbucket.org/tnguyenv/" target="_blank">My Bitbucket</a> &nbsp; &nbsp;
                <i class="fas fa-link" style="color: #0b84e0"></i>
                    <a href="https://bitbucket.org/biomodels/" target="_blank">BioModels on Bitbucket</a> &nbsp; &nbsp;
                <i class="fas fa-link" style="color: #0b84e0"></i>
                    <a href="https://itersdesktop.com" target="_blank">ITersDesktop</a> &nbsp; &nbsp;
                <i class="fas fa-map-marker-alt" style="color: #0b84e0"></i> Cambridge, United Kingdom
            </p>

        </div>
        <div class="col-md-2">
            <img class="rounded-circle shadow-4" src="assets/img/tung-square.jpg"
                    style="width: 150px; padding: 1vh;" alt="Profile Picture"/>
        </div>
    </div>
    <div class="row">
        <div class="col-md-9">
            <div id="summary">
                <h3 class="border-bottom">Summary</h3>
                <p>With over 20 years of professional experience, I'm driven to leverage 
my expertise in full-stack
development, front-end technologies and cloud solutions to drive product innovation. 
Enthusiastic, good work ethics, reliable.
Independently and professionally manages BioModels
production system across the whole, large code base. Good
task prioritization. Proactive. Very approachable, always
eager to support database curators or external users.
Excellent work in preparing an archival instance of
BioModels in BioStudies, including good collaboration with
BioStudies team. In the process also intensified
collaboration with other EBI resources like EuropePMC,
identifiers.org, and Reactome.
                </p>
            </div>
            <div id="experience">
                <h3 class="border-bottom">Experience</h3>
                <p><span class="title">Senior Software Engineer</span><br/>
                    <i class="fas fa-building"></i> <span class="organisation"> EMBL-EBI</span><span>&nbsp;<i
                            class="fas fa-calendar-alt"></i> Oct
                        2019 — Sep 2025 <i class="fas fa-map-marker-alt"></i> Cambridge, United Kingdom</span></p>
                <ul class="details">
                    <li>Cloud solutions: built CI/CD pipelines with GitLab; migrated BioModels services and its 
                        legacies (in PHP, Node.js, MongoDB, NoSQL, etc.) to EBI's Kubernetes cloud.</li>
                    <li>Microservices: a service with Spring Boot 2.0 to create CombineArchive files,
                        a service using Flask to download and upload large files over 1GB, a service using
                        FastAPI combining to Slurm to
                        exchange the requests between BioModels and EBI's High-Performance Computing (HPC) Cluster.</li>
                    <li>Extended REST API to allow programmatic submissions and supported our 
                        collaborators to update over 1000 curated models using the API.</li>
                    <li>Developed Python pipelines to submit BioModels content to BioStudies as a mirror repository. 
                        Worked with GitHub actions, huge data submissions, Redis cache, Bash script. 
                    </li>
                    <li>Refactored the whole, large code base of BioModels to speed up Kubernetes cloud deployment, 
                        reducing from 12 to 3 minutes.</li>
                    <li>Technologies updated: React, Vue (worked with BioStudies), MongoDB.</li>
                    <li>Always go the extra mile when supporting end users, collaborators, and our database curators. </li>
                </ul>
                <p><span class="title">Software Engineer</span><br/>
                    <i class="fas fa-building"></i><span class="organisation"> EMBL-EBI</span> <span>&nbsp;<i
                            class="fas fa-calendar-alt"></i> Nov 2015 — Sep 2019 <i
                            class="fas fa-map-marker-alt"></i> Cambridge, United
                        Kingdom</span></p>
                <ul class="details">
                    <li>Primarily upgraded BioModels using JUMMP written in Grails Groovy and Java.
                        Maintained the existing business implementations and added new features.</li>
                    <li>Model Repositories: support various community standards and formats for computational models
                    such as SBML, PharmML, MDL, etc. Worked with Ontologies, Metadata RDF, FAIRness.
                    <li>Searching and High-Performance Computing (HPC): Solr, BioSolr, indexing-based search, IBM Spectrum LSF
                    <li>Programming: Spring, Java, Grails 2.x, Groovy, MySQL, LDAP, Bash script
                    <li>Agile Development, Git, JIRA
                    <li>JavaScript, jQuery, Backbone, Underscore, ReactJS, Node, AngularJS 1.0
                    <li>FrontEnd technologies: Angular 8, Foundation CSS Framework 5.0, 6.0, Bootstrap 4, 5
                    <li>Taken part in building a model repository using an open source project, called JUMMP, a
                    deliverable of the DDMoRe project</li>
                </ul>

                <p><span class="title">PhD Student</span><br/>
                    <i class="fas fa-building"></i> <span class="organisation">LaBRI</span> <span>&nbsp;<i
                            class="fas fa-calendar-alt"></i> Oct 2010 — Feb 2015 <i
                            class="fas fa-map-marker-alt"></i> Bordeaux,
                    France</span></p>
                <ul class="details">
                    <li>Investigated and proposed a potential approach to analyse the enormous sets of elementary flux
                        modes in metabolic plant cell networks, then integrate the analysed results for finding motifs:
                        employed Minimal Cut Sets into analysing feasible pathways.
                        Studied methods of complex network analysis, clustering algorithms.</li>
                    <li>Built and modelled biological graphs. Researched solutions for the project and completed the
                        final report as the doctoral thesis.</li>
                    <li>Participated in conferences, workshops and gave two talks on related works.</li>
                    <li>Used C++, Python, and R programming language massively.</li>
                </ul>
                <p><span class="title">Software Developer — Trainee</span><br/>
                    <i class="fas fa-building"></i> <span class="organisation">LaBRI</span> <span>&nbsp;<i
                            class="fas fa-calendar-alt"></i> Jun 2019 — Nov 2019 <i
                            class="fas fa-map-marker-alt"></i> Bordeaux,
                    France</span></p>
                <ul class="details">
                    <li>Researched and implemented a Graph Editor in 3D using Common Lisp programming
                        language to manage metabolic networks. It allows users to visualise graphs 
                        in 3D and interact with its graphical elements.</li>
                </ul>
                <p><span class="title">University Lecturer</span><br/>
                    <i class="fas fa-building"></i> <span class="organisation">Hoa Sen University</span> <span>&nbsp;<i
                            class="fas fa-calendar-alt"></i> Aug 2004 — Oct 2010 <i
                            class="fas fa-map-marker-alt"></i> Ho Chi Minh City,
                        Vietnam</span></p>
                <ul class="details">
                    <li>
                    Be responsible for the syllabuses of the courses about Microsoft Office suite, 
                    the courses about Web Design with Dreamweaver, Presentation Design 
                    with PowerPoint. Delivered these courses as well as supervised students.</li>
                    <li>Designed and delivered new Software Engineering courses: <em>Software Architecture using
                        Design Patterns</em>
                        and
                        <em>Usability Engineering</em>.</li>
                    <li>Supervised research groups, supported students through a pastoral or advisory role.</li>
                    <li>Participated in staff training activities; Published a book <strong>Microsoft Excel 2010</strong> in 2010 in Vietnamese.</li>
                    <li>Webmaster for Department of Maths and Information Technology’s website.</li>
                </ul>
            </div>
            <div id="projects">
                <h3 class="border-bottom">Projects</h3>
                <p><span class="title"><a href="https://www.ddmore.foundation/" target="_blank">DDMoRe</a></span>
                    (A <b>D</b>rug<b> D</b>isease <b>Mo</b>del <b>Re</b>sources)<br/>
                    <span class="d-none d-sm-block">Maintain and enhance the published open-source DDMoRe standards and tools to deliver and
                    improve the quality, efficiency and cost-effectiveness of Model-Informed Drug Discovery &
                    Development (MID3) and Therapeutic Use. I contributed to developing and delivering its <a
                            href="https://www.ddmore.foundation/model-repository/" target="_blank">model
                        repository</a> and it is an open source project named
                    <a href="https://bitbucket.org/jummp/jummp/wiki/Home" target="_blank">JUMMP</a>
                    (<b>JU</b>st a <b>M</b>odel <b>M</b>anagement <b>P</b>latform).</span>
                </p>
                <p><span class="title">
                    <a href="https://www.ebi.ac.uk/biomodels" target="_blank">BioModels</a> using JUMMP</span><br/>
                    <span class="d-none d-sm-block">A repository of mathematical models of biological and biomedical systems.
                    It hosts a vast
                    selection of existing literature-based physiologically and pharmaceutically relevant mechanistic
                    models in standard formats.
                    Our mission is to provide the systems modelling community with reproducible,
                    high-quality, freely accessible models published in the scientific literature.</span>
                </p>
                <p><span class="title"><a href="https://bitbucket.org/biomodels/deepoc/src/master/" target="_blank">DeepOC</a></span><br/>
                <span class="d-none d-sm-block">Developed a Deep Learning model to classify the models 
                    deposited in BioModels based on their Gene Ontology information using Deep Learning.
                    The deliverables are a Deep Learning model implemented in Tensorflow and an 
                    <a href="https://d3js.org/" target="_blank">D3</a> zoomable sunburst chart to facilitate
                    browsing models based classes, automating the classification and predicting new classes.</span>
                </p>
            </div>
        </div>
        <div class="col-md-3">
            <div id="education">
                <h3 class="border-bottom">Education</h3>
                <p><span class="title">PhD in Computer Science</span><br/>
                    <i class="fas fa-building"></i> <span class="organisation">Universit&#233; de Bordeaux, </span>
                    <i class="fas fa-calendar-alt"></i> Oct 2010 — Feb 2015
                    <i class="fas fa-map-marker-alt"></i> Talence, France
                </p>

                <p><span class="title">Master's in Software Engineering</span><br/>
                    <i class="fas fa-building"></i> <span class="organisation">Universit&#233; de Bordeaux</span>
                    <i class="fas fa-calendar-alt"></i> Sep 2017 — Dec 2019
                    <i class="fas fa-map-marker-alt"></i> Ho Chi Minh City, Vietnam
                </p>

                <p><span class="title">Bachelor's in Computer Science</span><br/>
                    <i class="fas fa-building"></i> <span class="organisation">University of Quy Nhon</span>
                    <i class="fas fa-calendar-alt"></i> Sep 2000 — Jun 2004
                    <i class="fas fa-map-marker-alt"></i> Quy Nhon, Vietnam</p>
            </div>
            <!--<div id="achievements">
                <h3 class="border-bottom">Achievements</h3>
            </div>-->
            <div id="skills">
                <h3 class="border-bottom">Skills</h3>
                <p>Agile Methodologies, JIRA, Exhibit Leadership and mentorship, Problem-solving, Time
                    Management, Priority, Teamwork, Communication, Software Development, Software Architecture
                    Design, Debugging, Microservices architecture, Technological literacy, Organisation,
                    Adaptability, Empathy, Data analysis, Dependability</p>
            </div>
            <div id="certifications">
                <h3 class="border-bottom">Certifications</h3>
                <span><a
                        href="https://www.coursera.org/account/accomplishments/verify/W364QC9K4P7X" target="_blank">Application Development using Microservices and Serverless
                </a><br/>(Issued on Dec 27, 2024 by Coursera)</span><br/>Advanced and detailed study of creating
                microservices using various methodologies and running the applications
                on the IBM Cloud Code Engine with Openshift.<br/>
                <span><a href="https://www.coursera.org/account/accomplishments/verify/A5P8D0ELDPR7"
                         target="_blank">Continuous Integration and Continuous Delivery (CI/CD)
</a><br/>(Issued on Nov 23, 2024 by Coursera)</span><br/>
                Revised the key features of CI, explore social coding, and the Git Feature Branch Workflow. Grasped
                how to build and run a CI/CD pipeline to an OpenShift Kubernetes cluster, also
                discovered standard CD
                tools.<br/>
                See more certifications: <a href="https://tinyurl.com/tomcerts"
                    target="_blank">https://tinyurl.com/tomcerts</a>
            </div>
            <div id="languages">
                <h3 class="border-bottom">Languages</h3>
                <div class="row">
                    <div class="col col-sm-12 col-md-6">
                        <div class="row">
                            <div class="col-12 col-sm-12 col-md-6">
                                <strong>Vietnamese</strong><br/>
                                Native
                            </div>
                            <div class="col-12 col-sm-12 col-md-6">
                                <i class="fas fa-star"></i>
                                <i class="fas fa-star"></i>
                                <i class="fas fa-star"></i>
                                <i class="fas fa-star"></i>
                                <i class="fas fa-star"></i>
                            </div>
                        </div>
                    </div>
                    <div class="col col-sm-12 col-md-6">
                        <div class="row">
                            <div class="col-12 col-sm-12 col-md-6">
                                <strong>English</strong><br/>
                                Fluent
                            </div>
                            <div class="col-12 col-sm-12 col-md-6">
                                <i class="fas fa-star"></i>
                                <i class="fas fa-star"></i>
                                <i class="fas fa-star"></i>
                                <i class="fas fa-star"></i>
                                <i class="fa-solid fa-star-half-stroke"></i>
                            </div>
                        </div>
                    </div>

                </div>
                <div class="row">
                    <div class="col col-sm-12 col-md-6">
                        <div class="row">
                            <div class="col-12 col-sm-12 col-md-6">
                                <strong>French</strong><br/>
                                Lower Inter
                            </div>
                            <div class="col-12 col-sm-12 col-md-6">
                                <i class="fas fa-star"></i>
                                <i class="fas fa-star"></i>
                                <i class="fa-solid fa-star-half-stroke"></i>
                                <i class="far fa-star"></i>
                                <i class="far fa-star"></i>
                            </div>
                        </div>
                    </div>
                    <div class="col col-sm-12 col-md-6">
                        <div class="row">
                            <div class="col-12 col-sm-12 col-md-6">
                                <strong>Spanish</strong><br/>
                                Beginner
                            </div>
                            <div class="col-12 col-sm-12 col-md-6">
                                <i class="fas fa-star"></i>
                                <i class="fa-solid fa-star-half-stroke"></i>
                                <i class="far fa-star"></i>
                                <i class="far fa-star"></i>
                                <i class="far fa-star"></i>
                            </div>
                        </div>
                    </div>

                </div>
            </div>
            <div id="hobbies">
                <h3 class="border-bottom">Hobbies</h3>
                <div class="row">
                    <div class="col">
                        <span style="font-weight: bolder">Badminton</span><br/>
                        <span>Playing in the regional league as my best strong sport</span>
                    </div>
                    <div class="col">
                        <span style="font-weight: bolder">Swimming</span><br/>
                        <span>Going to swimming pool every Tuesday with my family</span>
                    </div>
                </div>
            </div>
            <div id="references">
                <h3 class="border-bottom">References</h3>
                <div class="row">
                    <div class="col">
                        <span style="font-weight: bolder">Prof Dr Beurton-Aimar Marie</span><br/>
                        <span>My PhD supervisor, LaBRI, Université de Bordeaux, France</span>
                    </div>
                    <div class="col">
                        <span style="font-weight: bolder">Henning Hermjakob</span><br/>
                        <span>Group Leader, Molecular Networks, EMBL-EBI, United Kingdom</span>
                    </div>
                </div>
                <div class="row">
                    <div class="col">
                        <span style="font-weight: bolder">Dr Rahuman S Malik-Sheriff</span><br/>
                        <span>Project Leader, EBML-EBI, United Kingdom</span>
                    </div>
                    <div class="col">
                        <span style="font-weight: bolder">Mihai Glont</span><br/>
                        <span>Tech Lead, Featurespace, VISA Inc., United Kingdom</span>
                    </div>
                </div>
                <div class="row">
                    <div class="col">
                    <p>(<strong>Note</strong>: Details of the references are available upon request.)</p></div>
                </div>
            </div>
        </div>
    </div>
    <div class="row">
        <div class="col">
            <p class="border-top" style="font-size: smaller;">
                <span style="font-style: italic;">Online version of this CV is available at</span> 
            <a href="https://ntung.github.io/curriculum-vitae" target="_blank">
                https://ntung.github.io/curriculum-vitae</a>
            </p>
        </div>
    </div>
</div>
<!-- javascript -->
<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.6/dist/js/bootstrap.bundle.min.js"
        crossorigin="anonymous"></script>
<script src="assets/javascript/common.widgets.js"></script>
</body>
</html>
